distribution/docs/src/main/resources/_contents/_apis/migration-api-contents.adoc

```diff
@@ -0,0 +1,104 @@
+
+The contract for a `migratable` is stored here. It is used by both sub-interfaces `ConfigurationMigratable` and `DataMigratable`.
+
+[WARNING]
+====
+Do not implement `Migratable` directly; it is intended for use only as a common base interface.
+Instead, the appropriate sub-interface should be used.
+====
+
+The `org.codice.ddf.migration.Migratable` interface defines these methods:
+
+.`MigrationMetadata export(Path exportPath) throws MigrationException`
+.`String getDescription()`
+.`boolean isOptional()`
+
+The `exportPath` in `export(Path exportPath)` is the path where all of the exportable data is copied.
+It is provided via an argument to the `migration:export` console command or via the Export Dialog in the ${admin-console}.
+The default value is `<${branding}_HOME>/etc/exported`.
+It is the responsibility of a `Migratable` to prevent naming collisions upon export.
+For example, if a `Migratable` writes files for its export, it must namespace the files.
+The `getDescription()` operation returns a short description of the type of data exported by the `Migratable`.
+The `isOptional()` operation returns whether the exported data for the `Migratable` is optional or required.
+The description and optional flag are for display purposes in the ${admin-console}.
+
+In order to create a `Migratable` for a module of the system, the `org.codice.ddf.migration.Migratable` interface must be implemented and the implementation must be registered under the `org.codice.ddf.migration.Migratable` interface as an OSGI service in the OSGI service registry.
+Creating an OSGI service allows for the `org.codice.ddf.configuration.migration.ConfigurationMigrationManager` to lookup all implementations of `org.codice.ddf.migration.Migratable` and command them to export.
+
+====== `ConfigurationMigratable`
+
+This is the base interface that must be implemented by all bundles or features that need to export and/or import system related settings
+(e.g. bundle specific Java properties, XML or JSON configuration files) during system migration.
+The information exported must allow the new system to have the same configuration as the original system.
+Only bundle or feature specific settings need be handled. All configurations stored in OSGi's
+`ConfigurationAdmin` will automatically be migrated and do not need to be managed by implementors of this class.
+
+Also, any other data not related to the system's configuration and settings (e.g., data stored in Solr) should be handled by a different
+service that implements the `DataMigratable` interface, not by implementors of this class.
+
+====== `DataMigratable`
+
+This is the interface that must be implemented by all bundles or features that need to export and/or import data during system migration.
+The data is not mandatory for the system to come up (e.g., data stored in Solr) and doesn't affect the system's configuration,
+i.e., without this data, the new system will still have the same configuration as the original one.
```

Comment:
I think at this point, I can answer my own question and say that we do use commas after `e.g.` and `i.e.`. :)